src/preprocessing/geometries/io.jl

```diff
@@ -55,29 +55,36 @@ function load(fs::FileIO.Stream{FileIO.format"STL_BINARY"}; ELTYPE=Float64)
     vertices = fill(fill(zero(ELTYPE), SVector{3}), 3n_faces)
     normals = fill(fill(zero(ELTYPE), SVector{3}), n_faces)
 
+    load_data!(face_vertices, vertices, normals, io)
+
+    return TriangleMesh(face_vertices, normals, vertices)
+end
+
+function load_data!(face_vertices::Vector{Tuple{SVector{3, T}, SVector{3, T},
+                                                SVector{3, T}}},
+                    vertices, normals, io) where {T}
     i = 0
     while !eof(io)
-        normals[i + 1] = SVector{3, ELTYPE}(read(io, Float32), read(io, Float32),
-                                            read(io, Float32))
+        normal = (read(io, Float32), read(io, Float32), read(io, Float32))
+        normals[i + 1] = SVector{3, T}(normal...)
 
-        v1 = SVector{3, ELTYPE}(read(io, Float32), read(io, Float32), read(io, Float32))
-        v2 = SVector{3, ELTYPE}(read(io, Float32), read(io, Float32), read(io, Float32))
-        v3 = SVector{3, ELTYPE}(read(io, Float32), read(io, Float32), read(io, Float32))
+        v1 = (read(io, Float32), read(io, Float32), read(io, Float32))
```

Comment:
```suggestion
        v1 = SVector(read(io, Float32), read(io, Float32), read(io, Float32))
```
You should be able to just create an `SVector` from a tuple without specifying type and size, as that's already encoded in the type of the tuple.